src/Microsoft.AspNet.Diagnostics/ErrorPageMiddleware.cs

```diff
@@ -183,31 +189,102 @@ private StackFrame StackFrame(Chunk line)
 
             int lineNumber = line.ToInt32();
 
-            return string.IsNullOrEmpty(file)
-                ? LoadFrame(string.IsNullOrEmpty(function) ? line.ToString() : function, string.Empty, 0)
-                : LoadFrame(function, file, lineNumber);
+            if (string.IsNullOrEmpty(file))
+            {
+                return GetStackFrame(
+                    // Handle stack trace lines like
+                    // "--- End of stack trace from previous location where exception from thrown ---"
+                    string.IsNullOrEmpty(function) ? line.ToString() : function,
+                    file: string.Empty,
+                    startLineNumber: 0,
+                    endLineNumber: 0);
+            }
+            else
+            {
+                return GetStackFrame(function, file, lineNumber, lineNumber);
+            }
         }
 
-        private StackFrame LoadFrame(string function, string file, int lineNumber)
+        // make it internal to enable unit testing
+        internal StackFrame GetStackFrame(string function, string file, int startLineNumber, int endLineNumber)
```

Comment:
Do we need the `endLineNumber`? Seeing every call to this send the same line number for start and end.